drivers/i2c/i2c_mcux_flexcomm.c

```diff
@@ -168,7 +174,7 @@ static int mcux_flexcomm_transfer(const struct device *dev,
 		}
 
 		/* Wait for the transfer to complete */
-		k_sem_take(&data->device_sync_sem, K_FOREVER);
+		k_sem_take(&data->device_sync_sem, I2C_TRANSFER_TIMEOUT_MSEC);
```

Comment:
Wont the hardware I2C controller still be stuck?

Reply:
No, but only if this FSL_FEATURE_I2C_TIMEOUT_RECOVERY is enabled (you can follow the macro and see it clears all interrupts). If you want to not bail out on the first error, I2C_RETRY_TIMES=X will allow for X attempts.

Reply:
Could you add a note highlighting this feature to the help string for the Kconfig? I suspect other developers using this would also like to enable bus recovery using this macro.